piqe_ocp_lib/api/resources/ocp_machine_management.py

```diff
@@ -353,3 +354,128 @@ def is_machine_deleted(self, machine_name: str, timeout: int = TIMEOUT) -> bool:
                     logger.info("Machine is no longer here")
                     return True
         return False
+
+
+class OcpMachineHealthCheck(OcpBase):
+    """
+    OcpMachineHealthCheck class extends OcpBase and encapsulates all methods
+    related to managing Openshift Machine Health Check.
+    :param kube_config_file: A kubernetes config file.
+    :return: None
+    """
+
+    def __init__(self, kube_config_file=None):
+        super(OcpMachineHealthCheck, self).__init__(kube_config_file=kube_config_file)
+        self.api_version = "machine.openshift.io/v1beta1"
+        self.kind = "MachineHealthCheck"
+        self.machinehealthcheck = self.dyn_client.resources.get(api_version=self.api_version, kind=self.kind)
+        self.machineset = OcpMachineSet(kube_config_file=kube_config_file)
+        self.machine = OcpMachines(kube_config_file=kube_config_file)
+        self.node = OcpNodes(kube_config_file=kube_config_file)
+
+    def get_all_machine_health_checks(self) -> ResourceList:
+        """
+        Get all defined Machine health check
+        :return: MachineHealthCheck on success OR an empty list on failure
+        """
+        api_response = list()
+        try:
+            api_response = self.machinehealthcheck.get(namespace=MACHINE_NAMESPACE)
+        except ApiException as e:
+            logger.error("Exception while getting all Machine Health Checks: %s\n", e)
+            raise Exception(e)
+        return api_response
+
+    def get_machine_health_check(self, machine_health_check_name: str) -> ResourceInstance:
+        """
+        Get a Machine health check set by name
+        :param machine_health_check_name: (str) name of the machine set
+        :return: MachineHealthCheck object on success OR None on failure
+        """
+        api_response = None
+        try:
+            api_response = self.machinehealthcheck.get(name=machine_health_check_name, namespace=MACHINE_NAMESPACE)
+        except ApiException as e:
+            logger.error("Exception while getting Machine Health Check: %s\n", e)
+            raise Exception(e)
+        return api_response
+
+    def configure_machine_health_check(self, machine_health_check_name: str) -> bool:
```

Comment:
nitpick, 'configure' makes me think we are modifying an existing mhc, but we seem to be creating a new one, correct? in that case i think 'create_machine_health_check' would be a more appropriate name for this method.

Reply:
I agree I was also not sure if it's appropriate to name configure here.
But lets discuss on the create/delete/watch and decide.